### PR TITLE
blogpost for kotlin project setup

### PR DESCRIPTION
**Pull request by zpencer**
This is the blog post to let users know about the improved kotlin support.

### CODE REVIEW

**Comment by zpencer** [comment]
@hsaliak @carl-mastrangelo @ejona86 PTAL

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title: Kotlin Gradle projects and gRPC
+published: false
+permalink: blog/kotlin-gradle-projects
+author: Spencer Fang
```
Comment (carl-mastrangelo) [inline _posts/2018-05-24-kotlin-project-setup.md]: Optional: provide an author-link if you would like.

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title: Kotlin Gradle projects and gRPC
+published: false
+permalink: blog/kotlin-gradle-projects
```
Comment (carl-mastrangelo) [inline _posts/2018-05-24-kotlin-project-setup.md]: make sure to add a published date before you merge.

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+author: Spencer Fang
+company: Google
+company-link: https://www.google.com
+---
+
+# Kotlin Gradle projects and gRPC
+
+Did you know that gRPC Java now has out of box support for Kotlin projects built with Gradle? Kotlin is a modern statically typed language developed by JetBrains that targets the JVM and Android. It is generally easy for Kotlin programs to interoperate with existing Java libraries. We have added support to the [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin/releases) so that the generated Java libraries are automatically picked up by Kotlin. You can now add the protobuf-gradle-plugin to your Kotlin project, and use gRPC just like you would with a Java or Android project.
```
Comment (carl-mastrangelo) [inline _posts/2018-05-24-kotlin-project-setup.md]: s/modern statically/modern, statically/

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+Below are Kotlin and Kotlin Android project set up examples.
```
Comment (carl-mastrangelo) [inline _posts/2018-05-24-kotlin-project-setup.md]: optional: "Kotlin and Kotlin" reads oddly, consider rephrasing.

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 3 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 4 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+author: Spencer Fang
+author-link: https://github.com/zpencer
+company: Google
+company-link: https://www.google.com
+---
+
+# Kotlin Gradle projects and gRPC
+
+Did you know that gRPC Java now has out of box support for Kotlin projects built with Gradle? Kotlin is a modern, statically typed language developed by JetBrains that targets the JVM and Android. It is generally easy for Kotlin programs to interoperate with existing Java libraries. We have added support to the [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin/releases) so that the generated Java libraries are automatically picked up by Kotlin. You can now add the protobuf-gradle-plugin to your Kotlin project, and use gRPC just like you would with a typical Java project.
+
```
Comment (LisaFC) [inline _posts/2018-05-24-kotlin-project-setup.md]: Add &lt;!--more--&gt;

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+Below are project configuration examples for a JVM application and an Android application using Kotlin.
```
Comment (LisaFC) [inline _posts/2018-05-24-kotlin-project-setup.md]: The follow examples show you how to configure a project for a JVM application and an Android application using Kotlin.

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+### Kotlin gRPC client and server
+
+The full example can be found [here](https://github.com/grpc/grpc-java/tree/master/examples/example-kotlin).
+
+Configuring gRPC for a Kolin project is the same as configuring it for a Java project.
```
Comment (LisaFC) [inline _posts/2018-05-24-kotlin-project-setup.md]: Kolin -> Kotlin

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 5 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 6 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 7 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  }
+}
+
+dependencies {
+  compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
+  // The rest of the projects dep are added below, refer to example URL
+}
+
+protobuf {
+  // The normal gRPC configuration goes here, refer to example URL
```
Comment (hsaliak) [inline _posts/2018-05-24-kotlin-project-setup.md]: if this information is small enough, can we reproduce it here?

**Comment by hsaliak** [comment]
Approved -- with some minor suggestions.

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+permalink: blog/kotlin-gradle-projects
+author: Spencer Fang
+author-link: https://github.com/zpencer
+company: Google
+company-link: https://www.google.com
+---
+
+# Kotlin Gradle projects and gRPC
+
+Did you know that gRPC Java now has out of box support for Kotlin projects built with Gradle? Kotlin is a modern, statically typed language developed by JetBrains that targets the JVM and Android. It is generally easy for Kotlin programs to interoperate with existing Java libraries. We have added support to the [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin/releases) so that the generated Java libraries are automatically picked up by Kotlin. You can now add the protobuf-gradle-plugin to your Kotlin project, and use gRPC just like you would with a typical Java project.
```
[the remark this answers is not in the document]
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+<!--more-->
+The follow examples show you how to configure a project for a JVM application and an Android application using Kotlin.
```
[the remark this answers is not in the document]
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

File: _posts/2018-05-24-kotlin-project-setup.md (hunk 8 above)
Comment (zpencer) [reply _posts/2018-05-24-kotlin-project-setup.md]: Done

**Pull request merged**